ADR/ADR044-translate-api-response-at-request-time.md

```diff
@@ -0,0 +1,417 @@
+    }
+  },
+  "steps": [
+    {
+      "id": 33,
+      "position": 1,
+      "next_step_id": null,
+      "type": "question_page",
+      "data": {
+        "question_text": "Upload your file",
+        "hint_text": "",
+        "answer_type": "file",
+        "is_optional": false,
+        "answer_settings": null,
+        "page_heading": "Annual report",
+        "guidance_markdown": "Some guidance",
+        "is_repeatable": false
+      },
+      "translations": {
+        "cy": {
+          "question_text": "Llwythwch eich ffeil i fyny",
+          "page_heading": "Adroddiad blynyddol",
+          "guidance_markdown": "Peth arweiniad"
+        }
+      },
+      "routing_conditions": []
+    }
+  ]
+}
+```
+
+Or with translated fields stored as objects (as proposed in the unimplemented [ADR007: Store form text data as objects](ADR007-store-form-text-data-as-objects.md))
+
+```jsonc
+{
+  "form_id": "7",
+  "name": {
+    "en": "A form for testing file upload",
+    "cy": "Ffurflen ar gyfer profi uwchlwytho ffeiliau"
+  },
+  "submission_email": "david.biddle@digital.cabinet-office.gov.uk",
+  "privacy_policy_url": null,
+  "form_slug": "a-form-for-testing-file-upload",
+  "support_email": null,
+  "support_phone": null,
+  "support_url": null,
+  "support_url_text": null,
+  "declaration_text": null,
+  "question_section_completed": false,
+  "declaration_section_completed": false,
+  "created_at": "2025-03-27T12:05:49.395300Z",
+  "updated_at": "2025-07-29T14:35:44.423406Z",
+  "creator_id": 1,
+  "what_happens_next_markdown": null,
+  "payment_url": null,
+  "submission_type": "email",
+  "share_preview_completed": false,
+  "s3_bucket_name": null,
+  "s3_bucket_aws_account_id": null,
+  "s3_bucket_region": null,
+  "language": "en",
+  "start_page": 33,
+  "steps": [
+    {
+      "id": 33,
+      "position": 1,
+      "next_step_id": null,
+      "type": "question_page",
+      "data": {
+        "question_text": {
+          "en": "Upload your file",
+          "cy": "Llwythwch eich ffeil i fyny"
+        },
+        "hint_text": "",
+        "answer_type": "file",
+        "is_optional": false,
+        "answer_settings": null,
+        "page_heading": {
+          "en": "Annual report",
+          "cy": "Adroddiad blynyddol"
+        },
+        "guidance_markdown": {
+          "en": "Some guidance",
+          "cy": "Peth arweiniad"
+        },
+        "is_repeatable": false
+      },
+      "routing_conditions": []
+    }
+  ]
+}
+```
+
+These cases are essentially the same in how forms-runner deals with them.
+Benefits:
+
+- Few admin/api changes required
+  Disadvantages:
+- More runner logic, since we'd have to manage choosing the right translation and fallback behaviour in runner.
+- We'd get less benefit from the Mobility gem - we'd be using it for managing the columns and dealing with translations in admin, but we wouldn't be using its querying or fallback features in forms-runner.
```

Comment:
If the Mobility gem doesn't help us in forms-runner, how much benefit would using it with the suffixed columns offer over making the translatable columns have jsonb type? Dealing with the object per translatable field in forms-runner as opposed to the suffixed fields feels like it would be slightly easier and neater - but there might not be much difference.

Dealing with a jsonb type with translations per language feels like it would be fairly straightforward and not really warrant using a gem to help us, but are there particular benefits Mobility gives - for instance, does it make it easier to check that translations have been provided for all the translatable values in forms-admin?